# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/endian.h.patch
# Copyright (C) 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

fix udev:

ata_id.c:(.text+0xd1): undefined reference to `le16toh'

  - Rene Rebe <rene@exactcode.de>

--- dietlibc/include/endian.h.vanilla	2011-01-05 19:05:43.000000000 +0100
+++ dietlibc/include/endian.h	2011-01-05 19:08:08.000000000 +0100
@@ -55,4 +55,16 @@
 #define __WORDSIZE 32
 #endif
 
+#ifdef _BSD_SOURCE
+/* Conversion interfaces. */
+#include <byteswap.h>
+
+#if __BYTE_ORDER == __LITTLE_ENDIAN
+#define le16toh(x) (x)
+#else
+#define le16toh(x) __bswap_16 (x)
+#endif
+
+#endif
+
 #endif
